lib/datadog/appsec/waf.rb

```diff
@@ -648,9 +650,10 @@ def run(input, timeout = LibDDWAF::DDWAF_RUN_TIMEOUT)
           end
 
           # retain C objects in memory for subsequent calls to run
-          retain(input_obj)
+          retain(persistent_data_obj)
+          retain(ephemeral_data_obj)
```

Comment:
We don't need to retain it will be freed at the end of `ddwaf_run`

Reply:
fixed [eb0273d](https://github.com/DataDog/libddwaf-rb/pull/47/commits/eb0273d0cd7cc1965db2ee6638cf43d109747216)